schema: IIO

jic23 opened the pull request:
Kernel bindings changes to take advantage of this will go to relevant mailing list shortly.

There are a few open questions in the patch descriptions.  Biggest one is whether it is appropriate
to have the level of detail I have in the way of description etc.  It seems that most of the schema in
this repo has a lot less.

I'm not sure what your process is for this repo, but hopefully this is close enough.

robherring commented on schemas/iio/adc.yaml:
```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: GPL-2.0
```
Dual license. I want everything in dtschema BSD at least. To the point that I've added some schemas without their descriptions because I haven't gone thru the relicensing exercise.

jic23 replied:
License thing was accidental.  Guess I cut and paste from an older file in the kernel tree.

jic23 commented on schemas/iio/adc.yaml:
```diff
@@ -0,0 +1,40 @@
+
+title: Generic IIO bindings for ADC channels
+
+select: true
```
Better to keep this on in the kernel tree perhaps as files will need to reference it anyway?

I didn't think this one through as obvious it will blow up..

jic23 commented on schemas/iio/iio-consumer.yaml:
```diff
@@ -0,0 +1,54 @@
+      then only the phandle portion of the pair will appear.
+
+  io-channel-names:
+    $ref: /schemas/types.yaml#/definitions/string-array
```
Great.

jic23 commented:
Updated as per Rob's comments on first version.
I also dropped the adc.yaml file on basis it may make more sense to just have that in the
kernel source tree.

jic23 commented:
Open question left, though is separate issue!
Is "label" something that we can make globally of type string? 
It's a standard property for IIO devices and there is a patch under review on the kernel mailing list to make it property for IIO channels as well.

jic23 commented:
Final version hopefully.  Had forgotten that a mount matrix is highly likely to have duplicated values.

robherring commented:
> Open question left, though is separate issue!
> Is "label" something that we can make globally of type string?
> It's a standard property for IIO devices and there is a patch under review on the kernel mailing list to make it property for IIO channels as well.

I thought it was, but appears not. It should be added to schemas/dt-core.yaml.

robherring commented:
> Updated as per Rob's comments on first version.
> I also dropped the adc.yaml file on basis it may make more sense to just have that in the
> kernel source tree.

I prefer common schemas here, but haven't been pushing that too hard as review in the kernel is easier and there's the relicensing exercise.
